src/utils.ts

```diff
@@ -16,32 +33,137 @@ export const generateTSSEndpoints = (tssNodeEndpoints: string[], parties: number
   let storage: Storage;
   try {
-    storage = window[type];
-    storageExists = true;
-    storageLength = storage.length;
+    if (type === "localStorage") storage = window.localStorage;
+    else storage = window.sessionStorage;
+
     const x = "__storage_test__";
     storage.setItem(x, x);
     storage.removeItem(x);
+
     return true;
   } catch (error) {
-    return (
-      error &&
-      // everything except Firefox
-      (error.code === 22 ||
-        // Firefox
-        error.code === 1014 ||
-        // test name field too, because code might not be present
-        // everything except Firefox
-        error.name === "QuotaExceededError" ||
-        // Firefox
-        error.name === "NS_ERROR_DOM_QUOTA_REACHED") &&
-      // acknowledge QuotaExceededError only if there's something already stored
-      storageExists &&
-      storageLength !== 0
-    );
+    return false;
   }
 }
+
+// TODO think which conversion functions to keep and how to export them.
+
+/**
+ * Parses a JWT Token, without verifying the signature.
+ * @param token - JWT Token
+ * @returns Extracted JSON payload from the token
+ */
+export function parseToken(token: string) {
+  const base64Url = token.split(".")[1];
+  const base64 = base64Url.replace("-", "+").replace("_", "/");
+  return JSON.parse(window.atob(base64 || ""));
+}
+
+/**
+ * Converts a mnemonic to a factor key.
+ * @param tKey - An initialized tKey instance.
+ * @param shareMnemonic - The mnemonic to convert.
+ * @returns The factor key.
+ */
+export async function mnemonicToKey(tKey: ThresholdKey, shareMnemonic: string): Promise<BN> {
+  const factorKey = await (tKey.modules.shareSerialization as ShareSerializationModule).deserialize(shareMnemonic, "mnemonic");
+  return factorKey;
+}
+
+/**
+ * Converts an arbitrary string to a factor key over an elliptic curve.
+ * @param ec - The elliptic curve.
+ * @param shareMnemonic - The mnemonic to convert.
+ * @returns The factor key.
+ */
+export async function stringToKey(ec: EC.curve.base, s: string): Promise<BN> {
+  const buf = Buffer.from(s);
+  // TODO this is compatible with the implementation before, but instead would
+  // be better to hash to scalar instead?
+  const bn = new BN(buf);
+  return bn.mod(ec.n);
+}
+
+/**
+ * Refreshes TSS shares. Allows to change number of shares. New user shares are
+ * only produced for the target indices.
+ * @param tKey - Tkey instance to use.
+ * @param factorPubs - Factor pub keys after refresh.
+ * @param tssIndices - Target tss indices to generate new shares for.
+ * @param factorKeyForExistingTSSShare - Factor key for existing TSS share.
+ * @param signatures - Signatures for authentication against RSS servers.
+ */
+async function refreshTssShares(
+  tKey: ThresholdKey,
+  factorPubs: Point[],
+  tssIndices: number[],
+  factorKeyForExistingTSSShare: BN,
+  signatures: string[]
+) {
+  const { tssShare, tssIndex } = await tKey.getTSSShare(factorKeyForExistingTSSShare);
```

Comment:
why refreshTssShares function is not in tkey?

Reply:
This seems a wrapper of tkey's refresh tss function getting needed params
tkey's refresh tss take in tssshare , and index, whlie this take in factorkey and get tssShare and tssIndex from it.